### PR TITLE
Add completions-validation step to CI

### PR DESCRIPTION
**Pull request by saschagrunert**
This CI step checks if the completions need to be re-commited into the
repository due to a flag change.

Follow up of: https://github.com/cri-o/cri-o/pull/2835

### CODE REVIEW

**Comment by saschagrunert** [comment]
Indeed the completions were wrong (root vs rootless execution), I adapted the CI pipeline to verify that and changed the make targets to be more independent.

File: completions/fish/crio.fish [hunk 1]
```diff
@@ -56,14 +56,14 @@ complete -c crio -n '__fish_crio_no_subcommand' -f -l profile -d 'enable pprof r
 complete -c crio -n '__fish_crio_no_subcommand' -f -l profile-port -r -d 'port for the pprof profiler'
 complete -c crio -n '__fish_crio_no_subcommand' -f -l read-only -d 'setup all unprivileged containers to run as read-only (default: false)'
 complete -c crio -n '__fish_crio_no_subcommand' -f -l registry -r -d 'registry to be prepended when pulling unqualified images, can be specified multiple times (default: configured in /etc/containers/registries.conf)'
-complete -c crio -n '__fish_crio_no_subcommand' -l root -s r -r -d 'crio root dir (default: "/home/sascha/.local/share/containers/storage")'
```
Comment (saschagrunert) [inline completions/fish/crio.fish]: It was a bit egoistic to think that every Linux user has my name. :innocent:

Comment (rhatdan) [reply completions/fish/crio.fish]: Probably a safe bet that they don't.  ...

**Comment by rhatdan** [comment]
LGTM

**Comment by saschagrunert** [comment]
Rebased on top of the latest master branch to resolve the conflicts.

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tests

**Comment by haircommander** [comment]
@saschagrunert can you rebase to fix ci and merge conflicts?

**Comment by saschagrunert** [comment]
Rebased on top of the latest master branch.

**Comment by saschagrunert** [comment]
@haircommander I think we have to bring this in to not not let the completions re-generate on every `make`.

**Comment by saschagrunert** [comment]
@haircommander just a friendly ping, do you think we can get this in? :)

**Comment by haircommander** [comment]
hell yeah we can
/lgtm

**Pull request merged**